### PR TITLE
Add gradient background settings

### PR DESCRIPTION
**Pull request by KaichenWang**
**Why are these changes introduced?**

- Enable support for gradient backgrounds

**What approach did you take?**

- Add theme settings to allow gradient (CSS background property)
- Add gradient settings (type `text`) under `Theme settings` > `Theme` > `Colors` and pair with their respective "fallback" color settings:
  - Background 1 
  - Background 2
  - Accent 1
  - Accent 2
- Integrate gradient setting into existing color system
- Introduce `.gradient` utility class. When applied to an element, that element will be styled with a gradient background (if the gradient setting is not blank, otherwise the fallback solid color is used).

**Other considerations**

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=126297210902)
- [Editor](https://os2-demo.myshopify.com/admin/themes/126297210902/editor)
- [Zip](https://github.com/Shopify/dawn/files/7077714/Dawn-Gradients-3.zip)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: templates/index.json [hunk 1]
```diff
@@ -2,9 +2,6 @@
   "sections": {
```
Comment (KaichenWang) [inline templates/index.json:2]: - [ ] revert before merge

**Comment by Guillaumegranger1** [comment]
>  And then the only remaining thing that I see missing is the number in the cart badge. If I'm not mistaken, its color should be as follow:
>
> • Accent 1: Accent 1
> • Accent 2: Accent 2
> • Background 1: Solid button label
> • Background 2: Background 2
> • Inverse: Text

☝️ I'm testing the demo regarding the last point and it seems to be fixed now. [Screen recording](https://screenshot.click/30-20-gjfee-vo56b.mp4)

@KaichenWang  are we all set?

**Review by Guillaumegranger1 — APPROVED** [review]

**Review by chrisberthe — APPROVED** [review]

File: sections/header.liquid [hunk 2]
```diff
@@ -73,7 +73,7 @@
     <path d="M.865 15.978a.5.5 0 00.707.707l7.433-7.431 7.579 7.282a.501.501 0 00.846-.37.5.5 0 00-.153-.351L9.712 8.546l7.417-7.416a.5.5 0 10-.707-.708L8.991 7.853 1.413.573a.5.5 0 10-.693.72l7.563 7.268-7.418 7.417z" fill="currentColor">
   </symbol>
 </svg>
-<{% if section.settings.enable_sticky_header %}sticky-header{% else %}div{% endif %} class="header-wrapper{% if section.settings.show_line_separator %} header-wrapper--border-bottom{% endif %}">
+<{% if section.settings.enable_sticky_header %}sticky-header{% else %}div{% endif %} class="header-wrapper{% if section.settings.show_line_separator %} header-wrapper--border-bottom{% endif %} color-{{ section.settings.color_scheme }} gradient">
```
Comment (ludoboludo) [inline sections/header.liquid:76]: Nitpick: 
```suggestion
<{% if section.settings.enable_sticky_header %}sticky-header{% else %}div{% endif %} class="header-wrapper color-{{ section.settings.color_scheme }} gradient{% if section.settings.show_line_separator %} header-wrapper--border-bottom{% endif %}">
```

File: sections/image-with-text.liquid [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {{ 'component-image-with-text.css' | asset_url | stylesheet_tag }}
 
 <div class="image-with-text {% if section.settings.full_width %}image-with-text--full-width{% else %}page-width{% endif %} color-scheme-{{ section.settings.color_scheme }}">
-  <div class="image-with-text__grid color-{{ section.settings.color_scheme }}{% if section.settings.layout == 'text_first' %} image-with-text__grid--reverse{% endif %} grid grid--gapless grid--1-col grid--2-col-tablet">
+  <div class="image-with-text__grid color-{{ section.settings.color_scheme }}{% if section.settings.layout == 'text_first' %} image-with-text__grid--reverse{% endif %} grid grid--gapless grid--1-col grid--2-col-tablet gradient">
```
Comment (ludoboludo) [inline sections/image-with-text.liquid:4]:
```suggestion
  <div class="image-with-text__grid color-{{ section.settings.color_scheme }} grid grid--gapless grid--1-col grid--2-col-tablet gradient{% if section.settings.layout == 'text_first' %} image-with-text__grid--reverse{% endif %}">
```

File: sections/rich-text.liquid [hunk 4]
```diff
@@ -4,7 +4,7 @@
 <noscript>{{ 'component-rte.css' | asset_url | stylesheet_tag }}</noscript>
 <noscript>{{ 'section-rich-text.css' | asset_url | stylesheet_tag }}</noscript>
 
-<div class="rich-text{% if section.settings.full_width %} rich-text--full-width{% endif %} color-{{ section.settings.color_scheme }}">
+<div class="rich-text{% if section.settings.full_width %} rich-text--full-width{% endif %} color-{{ section.settings.color_scheme }} gradient">
```
Comment (ludoboludo) [inline sections/rich-text.liquid:7]:
```suggestion
<div class="rich-text color-{{ section.settings.color_scheme }} gradient{% if section.settings.full_width %} rich-text--full-width{% endif %}">
```

**Review by ludoboludo — COMMENTED** [review]
Looks pretty cool! We're going to see some interesting backgrounds over time I think ([CSS pattern gallery](https://projects.verou.me/css3patterns/))

A couple questions: 
- product modals don't have the gradient as a bg, is that intentional? I could see how we don't want to have them everywhere. 
- color scheme for the image banner isn't pulling from the gradient, should it ?

File: assets/base.css [hunk 5]
```diff
@@ -1258,7 +1263,6 @@ details[open] > .share-button__fallback {
 .share-button__button,
 .share-button__fallback button {
   cursor: pointer;
-  background: rgb(var(--color-background));
   border: none;
```
Comment (ludoboludo) [inline assets/base.css:1266]: That might have been there specifically for Safari as from what I can see it adds a default bg colour when none is mentioned: 

SO you can maybe re add it with a value of `transparent`

Comment (ludoboludo) [reply assets/base.css:1266]: Seems like on android too.

**Review by ludoboludo — COMMENTED** [review]
And then on collage, the placeholder for the collection has a background set to it, so it looks a bit different than the other two cards: 


Which highlights the fact that apparently the parent is a bit wider than the actual image: [screenshot](https://screenshot.click/21-08-ypfsx-uaxv7.png) but that can be a follow up PR/issue

**Comment by KaichenWang** [comment]
@ludoboludo 

> And then on collage, the placeholder for the collection has a background set to it, so it looks a bit different than the other two cards

### For collage section

The following blocks have a `color-scheme` setting and therefore have a background that is not identical to the page background:
- Collection
- Image

The other two types of blocks do not have a `color-scheme` setting and therefore inherit the page's background
- Product
- Video

File: config/settings_schema.json [hunk 6]
```diff
@@ -28,12 +28,22 @@
         "label": "t:settings_schema.colors.settings.colors_accent_1.label",
         "info": "t:settings_schema.colors.settings.colors_accent_1.info"
       },
+      {
+        "id": "gradient_accent_1",
+        "type": "color_background",
+        "label": "t:settings_schema.colors.settings.gradient_accent_1.label"
```
[the remark this answers is not in the document]
Comment (KaichenWang) [reply config/settings_schema.json:34]: Agreed. There's still some ongoing discussion regarding merchant-facing copy

File: locales/en.default.schema.json [hunk 7]
```diff
@@ -11,9 +11,15 @@
           "label": "Accent 1",
           "info": "Used for solid button background."
         },
+        "gradient_accent_1": {
+          "label": "Accent 1 gradient"
+        },
```
Comment (ludoboludo) [inline locales/en.default.schema.json:16]: The look of it is a bit cluttered. Would be nice to some kinda hierarchy/grouping. But for future polish/improvement. Also it looks like the look of the settings has been updated 🤔 
[screenshot](https://screenshot.click/21-08-vlcqw-qxm8k.png)

**Review by ludoboludo — APPROVED** [review]
🚀

File: locales/en.default.schema.json [hunk 8]
```diff
@@ -724,6 +736,24 @@
     "header": {
       "name": "Header",
       "settings": {
+        "color_scheme": {
+          "options__1": {
+            "label": "Accent 1"
```
Comment (chrisberthe) [inline locales/en.default.schema.json:741]: Not urgent: there's quite a few instances of these colours throughout Dawn and each setting has its own translations—we can clean this up by referencing the same translation.

**Review by chrisberthe — APPROVED** [review]

**Pull request merged**